### ISSUE TITLE
Tweak operation order to guard against irreversible operations being executed twice

### ISSUE THREAD

**Issue by cameel** [golemfactory/concent#938]
Currently there are many places in our code where we mix database changes with irreversible side-effects like blockchain operations, adding tasks to work queue or making external HTTP requests. The problem is that database changes can be rolled back in case of a failure while those side effects cannot. There's no way to make the whole operation atomic (i.e. all or nothing).

Consider the situation where we need to update subtask state in the database and issue a payment. We can do either of these things first and the order determines what's left if the operation fails in the middle. If we pay first then a failure means that a payment was performed but we have no record of it - we risk that we'll pay twice. If we commit database changes first then a failure means that we allowed a state transition without payment. While both outcomes are undesirable, the former is worse because it lets the client retry and potentially trigger more than one payment.

The goal of the task is to ensure that we always update and commit database state first. Only then we should trigger the irreversible side effects.

A complete solution for the problem of interrupted operations would be to keep a journal that would tell us which of the planned operations were actually performed. This is something to consider in the future but it's out of scope of this task.

**Comment by PaweuB** [comment]
Pivotal:
https://www.pivotaltracker.com/story/show/162197386

**Comment by rwrzesien** [comment]
@cameel What do you think about `handle_send_force_payment` handler and its usage of `bankster.settle_overdue_acceptances`? It is currently being done before other database changes, but the logic after depends of the outcome of that call.

**Comment by cameel** [comment]
It's fine because in this case Concent first checks the blockchain to see if there are any settlement payments made before. If there are, it won't accept `SubtaskResultsAccepted` messages from before that.

Also, as long as `settle_overdue_acceptances()` creates and commits `DatabaseClaim` before the payment, Concent won't touch the deposit reserved in that use case.

The database changes related to creating out-of-band responses are a separate thing. It's fine if we don't create them (or maybe not really **fine** but better than the alternative anyway; this task is only about damage control, not fixing the underlying issue).

Actually there is one big issue: Concent will see the payment on the blockchain only after some delay. But it's out of the scope of this issue. This problem existed before and it's really about how Concent tracks forced payments. I proposed fixing it on Rocket and it's no up to Golem team to decide if and when to fix it.
